Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -603,24 +601,24 @@ void HTMLInputElement::attribute_changed(FlyString const& name, DeprecatedString
                 set_checked(true, ChangeSource::Programmatic);
         }
     } else if (name == HTML::AttributeNames::type) {
-        m_type = parse_type_attribute(value);
+        m_type = parse_type_attribute(value.value_or(""));
     } else if (name == HTML::AttributeNames::value) {
-        if (value.is_null()) {
+        if (!value.has_value()) {
             if (!m_dirty_value) {
                 m_value = DeprecatedString::empty();
                 update_placeholder_visibility();
             }
         } else {
             if (!m_dirty_value) {
-                m_value = value_sanitization_algorithm(value);
+                m_value = value_sanitization_algorithm(*value);
                 update_placeholder_visibility();
             }
         }
     } else if (name == HTML::AttributeNames::placeholder) {
         if (m_placeholder_text_node)
-            m_placeholder_text_node->set_data(MUST(String::from_deprecated_string(value)));
+            m_placeholder_text_node->set_data(MUST(String::from_deprecated_string(value.value_or(""))));
     } else if (name == HTML::AttributeNames::readonly) {
-        handle_readonly_attribute(value);
+        handle_readonly_attribute(value.value_or(""));
```

Comment:
I think this is a boog - this needs to check for if the `value` is null or not in `handle_readonly_attribute`

Reply:
Thanks for catching that, I'll also do a sweep for DS -> DFS conversions.